ecs-agent/gmsacredclient/credentialsfetcherclient.go

```diff
@@ -62,6 +62,116 @@ type CredentialsFetcherResponse struct {
 	KerberosTicketPaths []string
 }
 
+// Credentials fetcher is a daemon running on the host which supports gMSA on linux
+type CredentialsFetcherArnResponse struct {
+	//lease id is a unique identifier associated with the kerberos tickets created for a container
+	LeaseID string
+	//path to the kerberos tickets created for the service accounts
+	KerberosTicketsMap map[string]string
+}
+
+// HealthCheck() invokes credentials fetcher daemon running on the host
+// to check the health status of daemon
+func (c CredentialsFetcherClient) HealthCheck(ctx context.Context, serviceName string) (string, error) {
+	if len(serviceName) == 0 {
+		return "service name should not be empty", status.Errorf(codes.InvalidArgument, "service name should not be empty")
+	}
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.HealthCheckRequest{Service: serviceName}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.HealthCheck(ctx, request)
+	if err != nil {
+		seelog.Errorf("credentials-fetcher daemon is inactive")
```

Comment:
How about adding a bit more context to this error message like "during healthcheck" somewhere? We also encourage using the `logger` package to log messages as it provides a way to add context and standardizes the logging.

Reply:
addressed the comment with more text in the log